pandas/plotting/_matplotlib/core.py

```diff
@@ -969,6 +967,12 @@ def _make_plot(self):
         else:
             c_values = c
 
+        # plot colorbar if
+        # 1. colormap is assigned, and
+        # 2.`c` is a column containing only numeric values
```

Comment:
Do we have a test case that hits if non-numeric values are supplied? If not can you add one?

Reply:
@WillAyd , yup, it's `test_scatter_with_c_column_name_with_colors`, where `c='species'`, and the column `species` contains `["r", "r", "g", "g", "b"]`